Switch deprecated script prepublish to prepublishOnly

sudo-suhas opened the pull request:
This switches deprecated script `prepublish` to `prepublishOnly`. 

Relevant article - http://blog.lholmquist.org/npm-prepublish-changes/

Closes #217

sudo-suhas commented:
I didn't realise that CI relied on running `prepublish` on install, for npm scripts `depcheck` and `depcheck-json`. Should I add a `prepare` script?

lijunle commented:
Sure, thank you!

2017年7月26日 12:02，"Suhas Karanth" <notifications@github.com>写道：

> I didn't realise that CI relied on running prepublish on install, for npm
> scripts depcheck and depcheck-json. Should I add a prepare script?
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/depcheck/depcheck/pull/218#issuecomment-317941194>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABPIdM1lQ-32Zd9ERvdsSC20cng4Z1_Jks5sRrpBgaJpZM4OjYvq>
> .
>

sudo-suhas commented:
`prepare` script does not run in older npm versions 🤦‍♂️ . What do you suggest? Update CI config to run `npm run compile` and `npm run component` ?

lijunle commented:
Hi, @sudo-suhas 

I am sorry that I am late to be back on this PR. I think remove the `prepare` script and write `npm run compile` and `npm run component` into CI config is a good idea.

Thank you for raising this!

sudo-suhas commented:
I have removed the `prepare` script and added the commands for running `compile` and `component`. I think it makes sense to keep the `prepublishOnly` script in the eventuality that the module is ever published manually.

lijunle commented on .travis.yml:
```diff
@@ -8,6 +8,7 @@ node_js:
   - 8
 
 script:
+  - npm run compile && npm run component
```
Could you please split this into two lines?

sudo-suhas replied:
All done.

lijunle commented on appveyor.yml:
```diff
@@ -12,6 +12,7 @@ install:
 test_script:
   - node --version
   - npm --version
+  - npm run compile && npm run component
```
Could you please split this into two lines?

lijunle commented:
Thank you @sudo-suhas !
